src/wallet/rpc/spend.cpp

```diff
@@ -1406,6 +1406,11 @@ RPCHelpMan sendall()
                 }
             }
 
+            // If this transaction is too large, e.g. because the wallet has many UTXOs, it will be rejected by the node's mempool.
+            if (tx_size.weight > MAX_STANDARD_TX_WEIGHT) {
+                throw JSONRPCError(RPC_WALLET_INSUFFICIENT_FUNDS, "Transaction too large.");
```

Comment:
I initially tracked what error code the same check was returning from `CreateTransactionInternal`.

I now see that the 2 paths to call `CreateTransaction` return a different error code:

**RPC_WALLET_INSUFFICIENT_FUNDS** < `SendMoney` < `CreateTransaction` < `CreateTransactionInternal`
https://github.com/bitcoin/bitcoin/blob/718304d222671f98d2335cd9b90a3022f62d7b21/src/wallet/rpc/spend.cpp#L159-L162

**RPC_WALLET_ERROR** < `FundTransaction` < `CreateTransaction` < `CreateTransactionInternal`
https://github.com/bitcoin/bitcoin/blob/718304d222671f98d2335cd9b90a3022f62d7b21/src/wallet/rpc/spend.cpp#L702-L704

I am not sure if I am missing something but it seems that there is ambiguity. Probably because `CreateTransactionInternal` includes a variety of checks that in the end need to be represented with a single error code.

For example, a failed fee estimation is returned differently based on the RPC
https://github.com/bitcoin/bitcoin/blob/718304d222671f98d2335cd9b90a3022f62d7b21/test/functional/wallet_fallbackfee.py#L27-L29

Nevertheless, I agree with you that in this case RPC_WALLET_ERROR is more appropriate. I will change it together with the needed rebase.